oauth2client/contrib/gce.py

```diff
@@ -86,31 +95,57 @@ class AppAssertionCredentials(AssertionCredentials):
     """
 
     @util.positional(2)
-    def __init__(self, scope='', **kwargs):
+    def __init__(self, scope='', service_account_email='default', **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
-            scope: string or iterable of strings, scope(s) of the credentials
-                   being requested. Using this argument will have no effect on
-                   the actual scopes for tokens requested. These scopes are
-                   set at VM instance creation time and won't change.
+            scope:
+                string or iterable of strings, scope(s) of the credentials
+                being requested. Using this argument will have no effect on
+                the actual scopes for tokens requested. These scopes are
+                set at VM instance creation time and won't change.
+            service_account_email:
+                the email for these credentials. This can be used with custom
+                service accounts, or left blank to use the default service
+                account for the instance. Usually the compute engine service
+                account.
         """
         if scope:
             warnings.warn(_SCOPES_WARNING)
-        # This is just provided for backwards compatibility, but is not
-        # used by this class.
-        self.scope = util.scopes_to_string(scope)
+
+        self._service_account_email = service_account_email
+        self._service_account_info = None
+        self._project_id = None
+
         self.kwargs = kwargs
 
         # Assertion type is no longer used, but still in the
         # parent class signature.
         super(AppAssertionCredentials, self).__init__(None)
-        self._service_account_email = None
+
+    @property
+    def scopes(self):
+        return self._retrieve_scopes(httplib2.Http().request)
+
+    @property
+    def service_account_email(self):
```

Comment:
So, a potential problem

1. Someone creates a `AppAssertionCredentials` object on instance `A` with default service account `b@example.com`,
2. Serializes it
3. Deserializes it on instance `C` with default service account `d@example.com` and no access to `b@example.com`


Should
a. `ValueError` raised the first time access token is requested (`get_access_token` requests access token for an inaccessible email, because the deserialized email is `b@example.com` which `C` doesn't have access to)
b. The credentials work fine, provided `d@example.com` has the authorization

I'm leaning towards (a), so objects should have effectively immutable email address, in which case we could dump `self._service_account_email` and just use a partial `self._service_account_info` object to represent the "default" service account.